### PR TITLE
Add concurrency to worker deployments

### PR DESCRIPTION
**Pull request by galal-hussein**
https://github.com/rancher/rke/issues/176

### CODE REVIEW

File: services/workerplane.go [hunk 1]
```diff
@@ -4,43 +4,38 @@ import (
...
-		}
-		// run kubelet
-		// only one master for now
-		if err := runKubelet(host, workerServices.Kubelet, healthcheckDialerFactory); err != nil {
-			return err
-		}
-		if err := runKubeproxy(host, workerServices.Kubeproxy, healthcheckDialerFactory); err != nil {
-			return err
-		}
+		host := host
```
Comment (moelsayed) [inline services/workerplane.go]: Why the reassignment here ?

Comment (galal-hussein) [reply services/workerplane.go]: this is related to this https://golang.org/doc/faq#closures_and_goroutines

File: services/workerplane.go [hunk 2]
```diff
@@ -4,43 +4,38 @@ import (
...
+		errgrp.Go(func() error {
+			if err := doDeployWorkerPlane(host, workerServices, nginxProxyImage, sidekickImage, healthcheckDialerFactory, controlHosts); err != nil {
```
Comment (alena1108) [inline services/workerplane.go]: just `return doDeployWorkerPlane(host, workerServices, nginxProxyImage, sidekickImage, healthcheckDialerFactory, controlHosts)`

File: services/workerplane.go [hunk 3]
```diff
@@ -4,43 +4,38 @@ import (
...
+		return err
+	}
+	// Deploy worker components on worker hosts
 	for _, host := range workerHosts {
-		// run nginx proxy
-		if !host.IsControl {
-			if err := runNginxProxy(host, controlHosts, nginxProxyImage); err != nil {
+		host := host
+		errgrp.Go(func() error {
+			if err := doDeployWorkerPlane(host, workerServices, nginxProxyImage, sidekickImage, healthcheckDialerFactory, controlHosts); err != nil {
```
Comment (alena1108) [inline services/workerplane.go]: Same here

File: services/workerplane.go [hunk 4]
```diff
@@ -4,43 +4,38 @@ import (
...
 	}
+	if err := errgrp.Wait(); err != nil {
+		return err
+	}
+	// Deploy worker components on worker hosts
 	for _, host := range workerHosts {
-		// run nginx proxy
-		if !host.IsControl {
-			if err := runNginxProxy(host, controlHosts, nginxProxyImage); err != nil {
+		host := host
```
Comment (alena1108) [inline services/workerplane.go]: can we name the object on the left side differently?

**Comment by alena1108** [comment]
LGTM

**Pull request merged**